src/ssh/cert.rs

```diff
@@ -319,10 +319,12 @@ impl Certificate {
             Err(_) => return Err(Error::UnexpectedEof),
         };
         let serial = u64::from_be_bytes(serial);
+        let mut key = pubkey.clone();
+        key.key_type = key_type.clone();
```

Comment:
Are you sure this is right? I'm checking right now, but this would make the public key type inside the certificate a certificate type which I don't recall ever seeing.

Reply:
I was also surprised by that, but it's what `ssh-keygen -s` does on my machine. Thinking back, I guess it's not 100% necessary, as the non-rsa certificates are working fine, so I guess this is not a strict requirement, just some weird thing openssh does

Reply:
I'm running your changes now trying to determine what is different. I definitely want to follow the spec so if you're right I definitely want to merge this.

Reply:
I also definitely could be wrong on my implementation here as RSA was mostly an after thought.